service/cloudconfig/cloud_config.go

```diff
@@ -1,43 +1,267 @@
 package cloudconfig
 
 import (
+	"github.com/giantswarm/certificatetpr"
 	k8scloudconfig "github.com/giantswarm/k8scloudconfig"
-)
+	"github.com/giantswarm/microerror"
 
-type MasterExtension struct{}
+	"github.com/giantswarm/azure-operator/service/key"
+)
 
-// Files allows files to be injected into the master cloudconfig. It is
-// currently empty because the certificates will be stored in Key Vault.
+// Files allows files to be injected into the master cloudconfig.
 func (me *MasterExtension) Files() ([]k8scloudconfig.FileAsset, error) {
-	return nil, nil
+	var newFiles []k8scloudconfig.FileAsset
+
+	getSecretsScript, err := me.getMasterSecretsScript()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	newFiles = []k8scloudconfig.FileAsset{
+		getSecretsScript,
+	}
+
+	return newFiles, nil
 }
 
 // Units allows systemd units to be injected into the master cloudconfig.
-// TODO Add unit for downloading certificates from Key Vault.
 func (me *MasterExtension) Units() ([]k8scloudconfig.UnitAsset, error) {
-	return nil, nil
+	unitsMeta := []k8scloudconfig.UnitMetadata{
+		{
+			AssetContent: getKeyVaultSecretsUnit,
+			Name:         "get-keyvault-secrets.service",
+			Enable:       true,
+			Command:      "start",
+		},
+	}
+
+	units, err := me.renderUnits(unitsMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return units, nil
 }
 
 // VerbatimSections allows sections to be embedded in the master cloudconfig.
 func (me *MasterExtension) VerbatimSections() []k8scloudconfig.VerbatimSection {
 	return nil
 }
 
-type WorkerExtension struct{}
+// getMasterSecretsScript returns the script for downloading the master TLS
+// certificates from Key Vault on startup.
+func (me *MasterExtension) getMasterSecretsScript() (k8scloudconfig.FileAsset, error) {
```

Comment:
Here we generate a struct that is rendered in the script. It gets the secret names and sets the file locations.